design/one-pager-k8s-native-providers.md

```diff
@@ -0,0 +1,562 @@
+   will statically provision an `ObjectStore` (`minio.rook.io`) in the target
+   cluster.
+1. The `ObjectStore` controller should write any required connection secrets to
+   a Kubernetes `Secret` in the Crossplane control cluster.
+1. The Minio stack should expose a `Provider` (`minio.crossplane.io`) kind that
+   can reference the aforementioned `Secret`. Create a `Provider` of this kind.
+1. The Minio stack should also expose `Bucket` and `BucketClass` resources that
+   reference a Minio `Provider`. A `Bucket` may now be provisioned either
+   dynamically using a `Bucket` claim type or the Minio `Bucket` kind directly.
+
+Though the Minio stack strategy is somewhat cumbersome in that multiple
+Crossplane stacks must be installed in order to provision consumable object
+storage using Rook, it is more in-line with the structure of the Crossplane
+stack isolation model. In contrast, using controller logic in the Rook stack to
+provision buckets would be a divergence from our desire to model provider APIs
+in a high-fidelity manner. **For this reason, either waiting for support to be
+natively provided by Rook or creating a separate Minio stack appears to be the
+sensible direction forward.**
+
+### Stack Bundles
+
+If a separate Minio stack were to be implemented as detailed above, it may
+improve user experience to provide the ability to install multiple stacks
+together. This could either be exposed through releasing bundled distributions
+(i.e. Rook with Minio / Ceph / EdgeFS / etc.) or via a [Crossplane CLI] command.
+
+### More Sophisticated Scheduling
+
+It is possible that we will want to be able to schedule Kubernetes-native
+infrastructure resources in a more intelligent manner than the currently
+proposed `providerRef` pattern. This could include implementing scheduling
+policies that allow for resources to be provisioned based on geography, resource
+requirements, etc. An initial scheduler implementation may look similar to the
+updated `KubernetesApplication` scheduler controller, as outlined in [crossplane
+#859]. Because the scheduling behavior for most Kubernetes-native infrastructure
+components will look very similar, a shared scheduler reconciler could be added
+to [crossplane-runtime]. In fact, the `KubernetesApplication` scheduler
+controller may also be able to make use of the shared `crossplane-runtime`
+scheduler reconciler after it is implemented.
+
+### Automatic Operator Installation
+
+For any Kubernetes-native provider, there must be an operator installed in the
+target cluster for the resources to be provisioned. However, it would be nice to
+be able to provision resources in a cluster where the operator is not currently
+installed. This could be implemented by checking if the operator exists in the
+target cluster in the managed reconciler's `Create` method, and installing it if
+not. This may not be part of initial implementation, but should definitely be
+tracked and supported.
```

Comment:
This almost makes me wonder whether some kind of distinct, optional, run-once `Bootstrap()` method would make sense in `resource.ExternalClient`, but I think that's something to ponder another day.

Reply:
I like this pattern, updated crossplaneio/crossplane-runtime#34 with reference